tools/cmaketobzl/cmaketobzl.go

```diff
@@ -271,6 +275,65 @@ func (e *eval) unsetVariable(args []string) {
 	}
 }
 
+// setProject sets the name of the project and corresponding CMake variables.
+// See https://cmake.org/cmake/help/latest/command/project.html
+func (e *eval) setProject(args []string) {
+	if len(args) == 0 {
+		log.Fatal("Missing required project name")
+	}
+	// TODO(shahms): code-injection handling.
+	name, args := args[0], args[1:len(args)]
+	e.v.Set("PROJECT_NAME", name)
+	if e.isTopLevel() {
+		e.v.Set("CMAKE_PROJECT_NAME", name)
+	}
+	e.v.Set("PROJECT_SOURCE_DIR", path.Join(e.ProjectRoot(), e.CurrentDirectory()))
+	e.v.Set("PROJECT_BINARY_DIR", path.Join(e.ProjectRoot(), e.CurrentDirectory()))
+	e.v.Set(name+"_SOURCE_DIR", path.Join(e.ProjectRoot(), e.CurrentDirectory()))
+	e.v.Set(name+"_BINARY_DIR", path.Join(e.ProjectRoot(), e.CurrentDirectory()))
+
+	keywords := []string{"VERSION", "DESCRIPTION", "HOMEPAGE_URL", "LANGUAGES"}
+	for _, keyword := range keywords {
+		if args[0] == keyword {
+			switch keyword {
+			case "VERSION":
+				e.setProjectVersionVars(name, strings.Split(args[1], "."))
+				fallthrough
+			case "DESCRIPTION", "HOMEPAGE_URL":
+				e.setProjectVars(name, args[0], args[1])
+				args = args[2:len(args)]
+				break
```

Comment:
This break and the one below at 308 exit their switch clause rater than the `for` loop. If you really did mean to exit the loop, you can use a label (see below). But in this case it may be simpler to just `return`.

```go
nextKeyword:
  for _, kw := range keywords {
    if args[0] == kw {
      switch kw {
      case "FOO":
        blah
        break nextKeyword
```

or words to that effect.

Reply:
Relatedly: If the intent was to exit the switch clause, I recommend dropping the explicit `break` since that is what will happen anyway in the absence of `fallthrough`.

Reply:
Yeah, the intent was just to exit the switch.  Cleaned up and clarified, thanks!